Use count property instead of length

gyoshev opened the pull request:
Resolves immutable.js deprecation warning.

Closes #3493

TooTallNate commented on client/lib/recommended-sites-store/store.js:
```diff
@@ -30,7 +30,7 @@ function receiveRecommendations( data ) {
 	const previousRecs = recommendations;
 	if ( data && data.blogs ) {
-		if ( data.blogs.length === 0 ) {
+		if ( data.blogs.count === 0 ) {
```
I believe that `count` must be a function, i.e. `count()`.

blowery replied:
yup https://facebook.github.io/immutable-js/docs/#/Seq/count

blowery replied:
but the thing i don't understand is how data.blogs is becoming an immutable instance...

blowery replied:
ah, it's not. the warning is being thrown further down, when we check the length of recommedations and set last page.

blowery commented on client/lib/recommended-sites-store/store.js:
```diff
@@ -42,7 +42,7 @@ function receiveRecommendations( data ) {
 
 		if ( recommendations !== previousRecs ) {
-			if ( recommendations.length >= MAX_RECOMMENDATIONS ) {
+			if ( recommendations.count >= MAX_RECOMMENDATIONS ) {
```
so this is the actual problem and should be `count()`

gyoshev replied:
Ah, good catch. I'll address this shortly; do you find it sensible to add
this code branch to the test suite, as it currently passes the tests?

On Tuesday, February 23, 2016, Ben Lowery <notifications@github.com> wrote:

> In client/lib/recommended-sites-store/store.js
> <https://github.com/Automattic/wp-calypso/pull/3502#discussion_r53831251>:
>
> > @@ -42,7 +42,7 @@ function receiveRecommendations( data ) {
> >  		recommendations = recommendations.union( fromJS( pruned ) );
> >
> >  		if ( recommendations !== previousRecs ) {
> > -			if ( recommendations.length >= MAX_RECOMMENDATIONS ) {
> > +			if ( recommendations.count >= MAX_RECOMMENDATIONS ) {
>
> so this is the actual problem and should be count()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Automattic/wp-calypso/pull/3502/files#r53831251>.
>


-- 
Alexander A. Gyoshev
Web Developer / User Interface Engineer
mobile: +359 888 255838
Sofia, Bulgaria
http://gyoshev.net/

blowery replied:
Sure, we're pretty light on tests for this store right now. Any help would be appreciated. :)

gyoshev commented:
I fixed the count call and opened up the store to make it more testable:
* exposed the maximum recommendations constant
* introduced a method to reset the store between tests

Now there are at least a few tests for the isLastPage condition.

Your feedback is welcome!

blowery commented on client/lib/recommended-sites-store/store.js:
```diff
@@ -5,13 +5,19 @@ import pick from 'lodash/pick';
+		isLastPage = false;
+	},
+	maxRecommendations() {
+		return 100;
```
Let's bump this over into `./constants`

blowery replied:
actually, it matches what we had and it's only used one place. Let it lie.

blowery commented:
Looks great, thanks! :ship:
